src/resolve/resolved-dns-query.c

```diff
@@ -42,28 +42,43 @@ static void dns_query_candidate_stop(DnsQueryCandidate *c) {
 
         assert(c);
 
+        /* Detach all the DnsTransactions attached to this query */
+
         while ((t = set_steal_first(c->transactions))) {
                 set_remove(t->notify_query_candidates, c);
                 set_remove(t->notify_query_candidates_done, c);
                 dns_transaction_gc(t);
         }
 }
 
+static DnsQueryCandidate* dns_query_candidate_detach(DnsQueryCandidate *c) {
```

Comment:
hmm, i think there's prior art in our codebase to call funcitons like this "xyz_unlink()", but doesn#t matter